spec/system/post_index_spec.rb

```diff
@@ -0,0 +1,105 @@
+require 'spec_helper'
+
+RSpec.describe 'Post index page', type: :feature do
+  before(:each) do
+    @user = User.create(
+      name: 'Mohi',
+      bio: 'Aspiring FullStack Dev',
+      photo: 'https://via.placeholder.com/150',
+      posts_counter: 2
+    )
+
+    @user2 = User.create(
+      name: 'nati',
+      bio: 'FullStack Dev',
+      photo: 'https://via.placeholder.com/150',
+      posts_counter: 2
+    )
+
+    @post1 = Post.create(
+      title: 'Test',
+      text: 'First Post',
+      comments_counter: 0,
+      likes_counter: 0,
+      author: @user
+    )
+
+    @post2 = Post.create(
+      title: 'Test 2',
+      text: 'Second Post',
+      comments_counter: 0,
+      likes_counter: 0,
+      author: @user2
+    )
+
+    @comment1 = Comment.create(
+      text: 'First Comment',
+      post: @post1,
+      author: @user2
+    )
+
+    @comment2 = Comment.create(
+      text: 'Second Comment',
+      post: @post1,
+      author: @user2
+    )
+
+    @like1 = Like.create(
+      post: @post1,
+      author: @user2
+    )
+  end
+
+  describe 'the post index page' do
+    it 'displays user profile picture' do
+      visit user_posts_path(@user)
+      expect(page).to have_css("img[src*='https://via.placeholder.com/150']")
+    end
+
+    it 'displays the user username' do
+      visit user_posts_path(@user)
+      expect(page).to have_content('Mohi')
+    end
+
+    it 'displays number of posts the user has written' do
+      visit user_posts_path(@user)
+      expect(page).to have_content('Number of posts: 3')
+    end
+
+    it 'displays how many likes a post has' do
+      visit user_posts_path(@user)
+      expect(page).to have_content('likes: 1')
+    end
+
+    it 'displays some of the posts body' do
+      visit user_posts_path(@user)
+      expect(page).to have_content('First Post')
+    end
+
+    it 'displays posts title.' do
+      visit user_posts_path(@user)
+      expect(page).to have_content('Test')
+    end
+
+    it 'displays the first comments on a post.' do
+      visit user_posts_path(@user)
+      expect(page).to have_content('First Comment')
+    end
+
+    it 'section for pagination if there are more posts than fit on the view' do
+      visit user_posts_path(@user)
+      expect(page).to have_content('pagination')
+    end
+
+    it 'displays how many comments a post has' do
+      visit user_posts_path(@user)
+      expect(page).to have_content('comments: 2')
+    end
+
+    it 'click on a post, it redirects to that post show page.' do
+      visit user_posts_path(@user)
+      click_link 'Test'
+      expect(page).to have_current_path(user_post_path(@user, @post1))
+    end
+  end
+end
```

Comment:
- [ ] Ten of the errors come from the post_index_spec.rb file, please 🙏🏽 verify the database records: Ensure that the User and Post records are created successfully before creating the Comment records. You can verify this by checking the database or adding debug statements to print the created records.

```bash
Failures:

  1) Post index page the post index page displays user profile picture
     Failure/Error:
       @comment1 = Comment.create(
         text: 'First Comment',
         post: @post1,
         author: @user2
       )
     
     ActiveModel::MissingAttributeError:
       can't write unknown attribute `author_id`
     # ./spec/system/post_index_spec.rb:35:in `block (2 levels) in <top (required)>'

  2) Post index page the post index page displays the user username
     Failure/Error:
       @comment1 = Comment.create(
         text: 'First Comment',
         post: @post1,
         author: @user2
       )
     
     ActiveModel::MissingAttributeError:
       can't write unknown attribute `author_id`
     # ./spec/system/post_index_spec.rb:35:in `block (2 levels) in <top (required)>'

  3) Post index page the post index page displays number of posts the user has written
     Failure/Error:
       @comment1 = Comment.create(
         text: 'First Comment',
         post: @post1,
         author: @user2
       )
     
     ActiveModel::MissingAttributeError:
       can't write unknown attribute `author_id`
     # ./spec/system/post_index_spec.rb:35:in `block (2 levels) in <top (required)>'

  4) Post index page the post index page displays how many likes a post has
     Failure/Error:
       @comment1 = Comment.create(
         text: 'First Comment',
         post: @post1,
         author: @user2
       )
     
     ActiveModel::MissingAttributeError:
       can't write unknown attribute `author_id`
     # ./spec/system/post_index_spec.rb:35:in `block (2 levels) in <top (required)>'

  5) Post index page the post index page displays some of the posts body
     Failure/Error:
       @comment1 = Comment.create(
         text: 'First Comment',
         post: @post1,
         author: @user2
       )
     
     ActiveModel::MissingAttributeError:
       can't write unknown attribute `author_id`
     # ./spec/system/post_index_spec.rb:35:in `block (2 levels) in <top (required)>'

  6) Post index page the post index page displays posts title.
     Failure/Error:
       @comment1 = Comment.create(
         text: 'First Comment',
         post: @post1,
         author: @user2
       )
     
     ActiveModel::MissingAttributeError:
       can't write unknown attribute `author_id`
     # ./spec/system/post_index_spec.rb:35:in `block (2 levels) in <top (required)>'

  7) Post index page the post index page displays the first comments on a post.
     Failure/Error:
       @comment1 = Comment.create(
         text: 'First Comment',
         post: @post1,
         author: @user2
       )
     
     ActiveModel::MissingAttributeError:
       can't write unknown attribute `author_id`
     # ./spec/system/post_index_spec.rb:35:in `block (2 levels) in <top (required)>'

  8) Post index page the post index page section for pagination if there are more posts than fit on the view
     Failure/Error:
       @comment1 = Comment.create(
         text: 'First Comment',
         post: @post1,
         author: @user2
       )
     
     ActiveModel::MissingAttributeError:
       can't write unknown attribute `author_id`
     # ./spec/system/post_index_spec.rb:35:in `block (2 levels) in <top (required)>'

  9) Post index page the post index page displays how many comments a post has
     Failure/Error:
       @comment1 = Comment.create(
         text: 'First Comment',
         post: @post1,
         author: @user2
       )
     
     ActiveModel::MissingAttributeError:
       can't write unknown attribute `author_id`
     # ./spec/system/post_index_spec.rb:35:in `block (2 levels) in <top (required)>'

  10) Post index page the post index page click on a post, it redirects to that post show page.
      Failure/Error:
        @comment1 = Comment.create(
          text: 'First Comment',
          post: @post1,
          author: @user2
        )
      
      ActiveModel::MissingAttributeError:
        can't write unknown attribute `author_id`
      # ./spec/system/post_index_spec.rb:35:in `block (2 levels) in <top (required)>'
```